test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/testapp/integration/auth/customizer/SecurityMatcherTest.kt

```diff
@@ -0,0 +1,117 @@
+import java.time.LocalDate
+
+@Import(SecurityMatcherTest.CustomizerConfiguration::class)
+@SpringBootTest(
+  webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+  // allow overriding of the ResourceServerConfigurationCustomizer bean definition
+  properties = ["spring.main.allow-bean-definition-overriding=true"],
+)
+class SecurityMatcherTest : IntegrationTestBase() {
+  @TestConfiguration
+  class CustomizerConfiguration {
+
+    // Define a customizer that only applies to certain endpoints using the custom security matcher
+    @Bean
+    fun customSecurityMatcherCustomizer() = ResourceServerConfigurationCustomizer {
+      // Add a custom security matcher which requires the ACTIVE_CASELOAD role
+      securityMatcher { paths = listOf("/protected-by-custom-security-matcher") }
+      anyRequestRole { defaultRole = "CUSTOM_SECURITY_MATCHER" }
+    }
+
+    // And this is the security filter that uses the security marcher customizer
+    @Bean
+    fun securityMatcherSecurityFilterChain(
+      http: HttpSecurity,
+      @Qualifier("customSecurityMatcherCustomizer") customizer: ResourceServerConfigurationCustomizer,
+    ): SecurityFilterChain = HmppsResourceServerConfiguration().hmppsSecurityFilterChain(http, customizer)
+
+    // We have to explicitly define a standard customizer - this gets the library default and is a backstop for requests not
+    // matching the security matcher, so would have the standard customizer for the app e.g. including unauthorized paths etc
+    @Bean
+    fun hmppsCustomizer() = ResourceServerConfigurationCustomizer {}
+
+    // And we also have to explicitly define the standard filter chain or the library will "get out of the way" and
+    // only use the custom resource server defined above
+    @Bean
+    fun hmppsSecurityFilterChain(
+      http: HttpSecurity,
+      @Qualifier("hmppsCustomizer") customizer: ResourceServerConfigurationCustomizer,
+    ): SecurityFilterChain = HmppsResourceServerConfiguration().hmppsSecurityFilterChain(http, customizer)
+  }
+
+  @Test
+  fun `should return OK for an unauthorized endpoint`() {
+    webTestClient.get()
+      .uri("/info")
+      .exchange()
+      .expectStatus()
+      .isOk
+  }
+
+  @Test
+  fun `should return unauthorized for a protected endpoint`() {
+    webTestClient.get()
+      .uri("/active-caseload")
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `should return forbidden without a role`() {
+    webTestClient.get()
+      .uri("/active-caseload")
+      .headers(setAuthorisation())
+      .exchange()
+      .expectStatus()
+      .isForbidden
+  }
+
+  @Test
+  fun `should return forbidden with wrong role`() {
+    webTestClient.get()
+      .uri("/active-caseload")
+      .headers(setAuthorisation(roles = listOf("ROLE_WRONG")))
+      .exchange()
+      .expectStatus()
+      .isForbidden
+  }
+
+  @Test
+  fun `should allow access to the protected endpoint`() {
```

Comment:
missing test for denying access to this endpoint - not clear that the role is working / being used